2-mandatory/3-financial-times.js

```diff
@@ -14,24 +20,48 @@ function potentialHeadlines(allArticleTitles) {
     (you can assume words will always be seperated by a space)
 */
 function titleWithFewestWords(allArticleTitles) {
-    // TODO
-}
+    let lengthOfTitles = []
+    for(let i in allArticleTitles){
+        lengthOfTitles.push(allArticleTitles[i].split(" ").length)
+    }
+    let minimumTitlesIndex = lengthOfTitles.indexOf(Math.min(...lengthOfTitles))
+    return allArticleTitles[minimumTitlesIndex]
+        
+    }
+
 
 /*
     The editor of the FT has realised that headlines which have numbers in them get more clicks!
     Implement the function below to return a new array containing all the headlines which contain a number.
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
 function headlinesWithNumbers(allArticleTitles) {
-    // TODO
-}
+    let output = []
+    for(let i in allArticleTitles){
+        
+        if(allArticleTitles[i].match(/[0-9]/g) !== null){
+            output.push(allArticleTitles[i])
+        }
+        
+    }
+    return output
+ }
 
 /*
     The Financial Times wants to understand what the average number of characters in an article title is.
     Implement the function below to return this number - rounded to the nearest integer.
 */
 function averageNumberOfCharacters(allArticleTitles) {
-    // TODO
+    let allIntegers = []
+    for(let i in allArticleTitles){
+        allIntegers.push(allArticleTitles[i].length)
+    }
+    let total = 0;
+    for(let i = 0; i < allIntegers.length; i++) {
+    total += allIntegers[i];
+    }
+    let average = total / allIntegers.length;
```

Comment:
You did great here, I would advise using new lines to help separate your code to help the readability. Split it up into chunks of separate logic.